seacrowd/sea_datasets/memolon/memolon.py

```diff
@@ -0,0 +1,140 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from seacrowd.utils.configs import SEACrowdConfig
+from seacrowd.utils.constants import Licenses, Tasks
+
+_CITATION = """\
+@inproceedings{buechel-etal-2020-learning-evaluating,
+    title = "Learning and Evaluating Emotion Lexicons for 91 Languages",
+    author = {Buechel, Sven  and
+      R{\"u}cker, Susanna  and
+      Hahn, Udo},
+    editor = "Jurafsky, Dan  and
+      Chai, Joyce  and
+      Schluter, Natalie  and
+      Tetreault, Joel",
+    booktitle = "Proceedings of the 58th Annual Meeting of the Association for Computational Linguistics",
+    month = jul,
+    year = "2020",
+    address = "Online",
+    publisher = "Association for Computational Linguistics",
+    url = "https://aclanthology.org/2020.acl-main.112",
+    doi = "10.18653/v1/2020.acl-main.112",
+    pages = "1202--1217",
+}
+"""
+
+_DATASETNAME = "memolon"
+
+_DESCRIPTION = """\
+MEmoLon is an emotion lexicons for 91 languages, each one covers eight emotional variables and comprises over 100k word entries. There are several versions of the lexicons, the difference being the choice of the expansion model.
+"""
+
+_HOMEPAGE = "https://zenodo.org/record/3756607/files/MTL_grouped.zip?download=1"
+
+_LICENSE = Licenses.MIT.value
+
+_URLS = {
+    _DATASETNAME: "https://zenodo.org/record/3756607/files/MTL_grouped.zip?download=1",
+}
+
+_SOURCE_VERSION = "1.0.0"
+
+_LANGUAGE_MAP = {"en": "English", "ceb": "Cebuano", "tl": "Tagalog", "id": "Indonesian", "su": "Sundanese", "jv": "Javanese", "ms": "Malay", "vi": "Vietnamese", "th": "Thai", "my": "Burmese", "zh": "Chinese"}
+
+_SUPPORTED_TASKS = [Tasks.EMOTION_CLASSIFICATION]
+
+
+def seacrowd_config_constructor(lang, schema, version):
+    if lang not in _LANGUAGE_MAP:
+        raise ValueError(f"Invalid lang {lang}")
+
+    if schema != "source" and schema != "seacrowd_text_multi":
+        raise ValueError(f"Invalid schema: {schema}")
+
+    return SEACrowdConfig(
+        name="memolon_{lang}_{schema}".format(lang=lang, schema=schema),
+        version=datasets.Version(version),
+        description="MEmoLon {schema} schema for {lang} language".format(lang=_LANGUAGE_MAP[lang], schema=schema),
+        schema=schema,
+        subset_id="memolon",
+    )
+
+
+class Memolon(datasets.GeneratorBasedBuilder):
```

Comment:
please add Dataset suffix here
```suggestion
class MemolonDataset(datasets.GeneratorBasedBuilder):
```

Reply:
Hi @jamesjaya thanks for reviewing my PR. 

Regarding the dataset suffix, if I understand correctly, the instruction is to name the class similar to the python script name but in CamelCase (see below for reference).

https://github.com/SEACrowd/seacrowd-datahub/blob/16116789ff8e0bb5a3c87387586fcae1393af648/templates/template.py#L104-L105

Reply:
Well, as a reviewer, I have a conflicting instruction to make sure that the class name has `Dataset` suffix. 

Quoting this document https://docs.google.com/document/d/1uVtus5MXeM2lf4MmlSGhSXIHZXVLCqoaK6nh6sMyC3w/edit
```
CamelCase for dataloader class name (and “Dataset” is contained in the suffix of the class name).
```

I'll defer to @SamuelCahyawijaya